zomes/blog/code/src/post_validations.rs

```diff
@@ -1,40 +1,89 @@
-// THIS IS NOT FINAL
-// There are several things missing and not worked out yet.
-// Like:
-//   - validate_* functions need to receive a complex struct as argument
-//     that holds the entry to be validated next to the sources,
-//     validation package etc.
-//   - return type of these callbacks should be Result<..>, right?
-//   - how to declare/write validation packages?
-//   - do we need a guarding macro like zome_fns!?
-//
-// This file is WIP and placeholder to be overwritten by future
-// PR after completing the design discussion and reaching consensus
-// about these specifics.
 use serde_json;
 
+/// We declare the structure of our entry type with this Rust struct.
+/// It will be checked automatically by the macro below, similar
+/// to how this happens with functions parameters and zome_functions!.
 #[derive(Serialize, Deserialize)]
 struct Post {
     content: String,
-    date_created: String
+    date_created: String,
+    allow_comments: bool,
 }
 
-#[no_mangle]
-pub extern "C" fn validate_commit(_offset: i32) -> i32{
-    0
-}
+/// We need a macro that wraps those validation functions similar to
+/// zome_functions!, but with different parameters.
+validations! {
+    /// We also need to distinguish between the validation of an entry
+    /// (here) and validation of links (below)
+    [ENTRY] validate_post {
+        /// This is an enum that needs to be added to the HDK and that tells
+        /// Holochain what validation package the source needs to send
+        /// to validating DHT nodes.
+        /// Should be something like:
+        ///
+        /// enum ValidationPackage {
+        ///    Entry,           //sending only the entry
+        ///    ChainEntries,    //sending all (public?) source chain entries
+        ///    ChainHeaders,    //sending all source chain headers
+        ///    ChainFull,       //sending the whole chain, entries and headers
+        ///    Custom(String),  //sending something custom
+        /// }
+        [ValidationPackage::Entry]
```

Comment:
@zippy: We need to be able to specify certain entry types that should only be included in the package (= not the whole chain).